src/fmnoise/coldbrew.clj

```diff
@@ -181,11 +195,14 @@
                               :else 3))
         fdefn (filter some? (list name docstring args prepost))]
     `(do
-      (def ^:private ~fname
-         (cached
-          (with-meta
-            (fn [~@cache-key]
-              ~@body)
-            ~cache-options)))
-      (defn ~@fdefn (~fname ~@cache-key))
-      (vary-meta ~name merge ~(meta name)))))
+       (def ^:private ~fname (cached
+                              (with-meta
+                                (fn [~@cache-key]
+                                  ~@body)
+                                ~cache-options)))
+       (defn ~@fdefn (~fname ~@cache-key))
```

Comment:
maybe should apply cache metadata taken from `(meta ~fname)` to this function object rather than applying it to var?

Reply:
Looks interesting!🤔  I didn't want to fight with all those variables, but it doesn't look as bad as I imagined.

I'm busy with other stuff this week, but on the next week I should be able to return into this pr.